modules/apps/web-experience/export-import/export-import-resources-importer/src/main/java/com/liferay/exportimport/resources/importer/internal/messaging/ResourcesImporterHotDeployMessageListener.java

```diff
@@ -73,6 +77,16 @@
 
 	@Activate
 	protected void activate(final BundleContext bundleContext) {
+		_bundleContext = bundleContext;
+
+		_messageQueue = new LinkedList<>();
+
+		ServiceTrackerFactory.open(
+			_bundleContext,
+			"(objectClass=com.liferay.portal.osgi.web.portlet.tracker." +
+				"internal.PortletTracker)",
+			new PortletTrackerServiceTrackerCustomizer());
+
 		_serviceTrackerMap = ServiceTrackerMapFactory.openSingleValueMap(
```

Comment:
The basic idea is to wait for all the portlets available at startup time to be activated/initialized properly before starting to import any lar file, otherwise we'll loose valid data. However if one portlet is not installed at all, it is totally valid and the import controller will automatically skip to import any data belongs to it. 

Regarding PortletTracker the idea was to start the resources importer and queue the incoming deployment requests until the tracker is not available, then process the queue. However PortletTracker is not exported from its bundle, so we need a workaround because I did not want to change that behavior only for this purpose.